tools/clang/lib/CodeGen/CGHLSLMS.cpp

```diff
@@ -5193,17 +5205,8 @@ void CGMSHLSLRuntime::EmitHLSLRootSignature(CodeGenFunction &CGF,
 
   DXASSERT(CGF.getLangOpts().RootSigMajor == 1,
            "else EmitHLSLRootSignature needs to be updated");
```

Comment:
Can we move this DXASSERT to the constructor along with the checks on RootSigMinor to keep these together?